genie_registry/clinical.py

```diff
@@ -175,6 +175,33 @@ def _check_int_year_consistency(
     return ""
 
 
+def _check_year_death_validity(clinicaldf: pd.DataFrame) -> str:
+    """
+    Check if YEAR_DEATH >= YEAR_CONTACT
+
+    Args:
+        clinicaldf: Clinical Data Frame
+
+    Returns:
+        Error message if YEAR_DEATH if invalid
+    """
+    error = ""
+    # Generate temp dataframe to handle datatype mismatch in a column
+    temp = clinicaldf.copy()
+    # Convert YEAR_DEATH and YEAR_CONTACT to numeric, coercing errors to NaN
+    temp["YEAR_DEATH"] = pd.to_numeric(temp["YEAR_DEATH"], errors="coerce")
+    temp["YEAR_CONTACT"] = pd.to_numeric(temp["YEAR_CONTACT"], errors="coerce")
+    check_result = np.where(
+        (pd.isna(temp["YEAR_DEATH"]) | pd.isna(temp["YEAR_CONTACT"])),
+        "N/A",
+        temp["YEAR_DEATH"] >= temp["YEAR_CONTACT"],
+    )
+    index = [",".join(str(idx)) for idx, i in enumerate(check_result) if i == "False"]
```

Comment:
Wondering why you're joining the indices via "," as a string? I think maybe you meant to do something like this:
`indices = [idx for idx, result in enumerate(check_result) if result is False]`

What I've found helpful in the past is breaking up the validation functions into two parts: one function for outputting the invalid row indices and one function for outputting the error message so you can run tests on the actual invalid results.